docs: add kubernetes prerequisites and deployment steps (#39)
openclarity/docs.openclarity.io: content/docs/vmclarity/getting-started/deploy-kubernetes/_index.md

---
title: Deploy on Kubernetes
weight: 110
---

## Prerequisites

> * Install a tool to run local Kubernetes clusters. Here, [Kind](https://kind.sigs.k8s.io/) is used as the default option for creating a local cluster.

Reviewer: I'd assume the visitor of this page would like to install VMClarity on a "real" cluster

```suggestion
* You need to have a Kubernetes cluster. To try things out first, you could use [Kind](https://kind.sigs.k8s.io/) to create a local cluster.
```

Author: will keep the tool requirement here and add creating a cluster as a first step as suggested below

Reviewer 2: the prerequisite is the K8s cluster itself
```suggestion
* Running Kubernetes cluster. In this example, [Kind](https://kind.sigs.k8s.io/) is used as the default option for managing a local cluster.
```

> * Install a package manager. Here, [Helm](https://helm.sh/) is used as the default installer.

Reviewer: What do you think about something like this?

```suggestion
* [Helm](https://helm.sh/), to install VMClarity via its [Helm chart](https://github.com/openclarity/vmclarity/tree/main/installation/kubernetes/helm/vmclarity).
```

## Deployment steps

To deploy VMClarity to your Kubernetes cluster, complete the following steps.

1. Ensure the Kubernetes cluster is up and running. If you're using kind, you can check the status of your clusters with the following command:

    ```shell
    kind get clusters

>     ````

Reviewer 2: lets make the first step actually booting up a new cluster (since we are already using kind)
`kind create cluster --name vmclarity-k8s-demo`

1. Use Helm to install VMClarity. Run the following command, making sure to replace installation/kubernetes/helm/vmclarity with the correct path to the VMClarity Helm chart:

    ```shell

>     helm install vmclarity installation/kubernetes/helm/vmclarity \

Reviewer: We should build and distribute the chart as an OCI chart for example, instead of pointing to the chart directory in the repo.

Reviewer: Oh, I just found it, it is `ghcr.io/openclarity/charts/vmclarity:latest`

        --namespace vmclarity --create-namespace \
        --set orchestrator.provider=kubernetes \
        --set orchestrator.serviceAccount.automountServiceAccountToken=true \

>         --set exploitDBServer.containerSecurityContext.enabled=false

Reviewer: It would be better if it worked with the default values, we should consider changing them so it can be installed with a simple helm install command, what do you think?

Author: I suggest we open an issue in the VMClarity repo and update here once the deployment steps are simplified, because for now these are the required params.

---
title: Deploy on Kubernetes
weight: 110
---

## Prerequisites

* Install a tool to run local Kubernetes clusters. Here, [Kind](https://kind.sigs.k8s.io/) is used as the default option for creating a local cluster.

> * Install [Helm](https://helm.sh/) to install the application.

Reviewer 2: ```suggestion
* [Helm](https://helm.sh/) to install VMClarity on Kubernetes.
```

    ```shell
    kind get clusters
    ````

1. Use Helm to install VMClarity. Run the following command:

    ```shell
    helm install vmclarity oci://ghcr.io/openclarity/charts/vmclarity --version 0.6.0 \ 
        --namespace vmclarity --create-namespace \
        --set orchestrator.provider=kubernetes \
        --set orchestrator.serviceAccount.automountServiceAccountToken=true \
        --set exploitDBServer.containerSecurityContext.enabled=false
    ```

1. Verify that all the VMClarity pods have been successfully deployed by executing the following command:

    ```shell
    kubectl get pods -n vmclarity
    ```

1. Wait until all pods are in the `Running` state or have completed their initialization.

1. Once the pods are ready, start port forwarding to access the VMClarity gateway service. Use the following command to forward traffic from your local machine to the cluster:

    ```shell
    kubectl port-forward -n vmclarity service/vmclarity-gateway 8080:80
    ````

1. Access the VMClarity UI by navigating to [http://localhost:8080/](http://localhost:8080/) in your web browser.

    <p align="center" width="100%">
        <img width="75%" src="/img/vmclarity-ui-1.png">

>     </p>

Reviewer 2: lets include cleanup step (includes 2 commands: helm to delete the chart, and kind to remove the cluster)
